### PR TITLE
Move all the legacy handling into transit

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

This PR moves all the legacy message handling into `transit` namespace, so the outermost "edge" of `status-react`.
Doing so has several benefits:
* All the legacy handling (sending&receiving) is in one place and easy to inspect
* Inside logic of the application can be now bit simpler and easier to reason about
* Less potential "traps" for anybody working on the message/command-message logic
* We can finally almost completely delete the legacy `content-type-command-request` from the project, keeping its use just for the legacy code in the `transit` namespace.

Please take your patience to review the code changes and do it with scrutiny.

### Testing notes (optional):
*  Please do all your testing with special test public/1-1 channels first, then proceed to make sure that known public channel like `#status` can be displayed correctly on the PR build and only post messages from the PR build there as a last thing if everything worked fine previously. Doing so will minimise the danger of breaking public communications with protocol changes. It should be the case for all protocol touching PRs from now and will be better described and documented in `Protocol compatibility practises ADR` (coming soon)
* Test that account with previous `/send` and `/request` message is OK upon upgrading to this PR build and all command messages are displayed correctly
* Test that communication between this PR and any other builds (including the oldest build you can get your hands on, providing it's newer then `0.9.22`) is fine, with special regard to command messages

status: ready

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

File: src/status_im/transport/message/transit.cljs [hunk 1]
```diff
@@ -44,14 +44,36 @@
...
 (deftype MessageHandler []
   Object
   (tag [this v] "c4")
   (rep [this {:keys [content content-type message-type clock-value timestamp]}]
-    (if (= content-type constants/text-content-type)
-      ;; append new content add the end, still pass content the old way at the old index
+    (condp = content-type
+      constants/text-content-type ;; append new content add the end, still pass content the old way at the old index
       #js [(:text content) content-type message-type clock-value timestamp content]
-      #js [content content-type message-type clock-value timestamp content])))
```
Comment (janherich) [inline src/status_im/transport/message/transit.cljs:54]: @cammellos I know we had a discussion about this last time (always appending content at the end, even when not strictly necessary in command type messages), but I decided to remove it for the following reason, after discussion with @yenda :
1. As I said, most if not all the modification will be done in the content map itself, so no positional shuffling anymore (or very unlikely)
2. We want to avoid repeated content in message record whenever possible, because we fought hard to limit our bandwidth consumption and we don't want to reverse it for reasons like developer convenience - of course this is highly subjective (which aspect is more important) and I'm happy to follow team decision in this regard, currently it's 2:1 for no duplicate payload (@janherich & @yenda vs @cammellos).

File: src/status_im/transport/message/transit.cljs [hunk 2]
```diff
@@ -44,14 +44,36 @@
   (rep [this {:keys [name profile-image address fcm-token]}]
     #js [name profile-image address fcm-token]))
 
+;; It's necessary to support old clients understanding only older, needlesy complex and verbose command content
```
Comment (cammellos) [inline src/status_im/transport/message/transit.cljs]: Thanks for letting us know you didn't like the previous implementation, but it would be more useful to add which version was the last one to run that code, so we can ditch the code once we are sure that no clients are using it :)

File: src/status_im/transport/message/transit.cljs (hunk 1 above)
Comment (cammellos) [reply src/status_im/transport/message/transit.cljs:54]: @janherich As mentioned in the dm, I don't mind, I don't fully agree, but it's the team decision and I respect that

**Review by cammellos — APPROVED** [review]

File: src/status_im/transport/message/transit.cljs (hunk 2 above)
Comment (janherich) [reply src/status_im/transport/message/transit.cljs]: Sure, I will add it to the comment, but I don't think it's particularly important as it's version newer then `0.9.20`, so we have to support it (indefinitely ?) according to last team discussion about version compatibility.
Neither is my rant, but you would have to see the old payload to understand my frustration :))

Comment (cammellos) [reply src/status_im/transport/message/transit.cljs]: I'd rather not see it if it prompted such a bad reaction :) 
I would still add the version number, as it is also for external devs, who might not care about supporting previous versions, and also you never know if we are going to revisit those policies at some point ;)

Comment (janherich) [reply src/status_im/transport/message/transit.cljs]: True, I removed the unnecessary emotions and added facts (exact version where command `content` was upgraded to newer, sparser version)

File: src/status_im/transport/message/transit.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/transport/message/transit.cljs:54]: @janherich @cammellos I was in favor of no duplication for the fix but now there is already a few versions that expect the text to be in content. If they are not able to properly read the messages received that would violate the forward compatibility policy

File: src/status_im/transport/message/transit.cljs [hunk 3]
```diff
@@ -44,14 +44,36 @@
...
+       command-path))
+
 (deftype MessageHandler []
   Object
   (tag [this v] "c4")
   (rep [this {:keys [content content-type message-type clock-value timestamp]}]
-    (if (= content-type constants/text-content-type)
-      ;; append new content add the end, still pass content the old way at the old index
+    (condp = content-type
+      constants/text-content-type ;; append new content add the end, still pass content the old way at the old index
```
Comment (yenda) [inline src/status_im/transport/message/transit.cljs]: please make a choice here either text-content-type or content-type-text to match content-type-command or command-content-type. Anyway personally I would prefer namespaced keywords but that would be for another PR

File: src/status_im/transport/message/transit.cljs (hunk 1 above)
Comment (janherich) [reply src/status_im/transport/message/transit.cljs:54]: @yenda The question is not about the messages of content type `text` (duplication necessary there at this time-point) but about `command` content-type, where we don't have to append content as a last item of array (as the "old" content at zero index was always a map a not interpretation change is needed there), but making it so makes the messages more regular because we can be sure that from now on, valid message content is always at the last index, opposed to:
* old style (string) content as a first and new style (map) content as a last items for the text messages
* map style content as a first item for the command messages

File: src/status_im/transport/message/transit.cljs (hunk 3 above)
Comment (janherich) [reply src/status_im/transport/message/transit.cljs]: Ok, I will update it.

**Comment by janherich** [comment]
@yenda Please review again, I already changed constant `text-content-type` to `content-type-text` so it follows the same patter as rest of the content types

**Review by yenda — APPROVED** [review]
I am ok with it for now but I think it would be better if compatibility was in it's own namespace in the future, we could take inspiration from realm migration layout and be able to easily figure out for which version each compatibility is for.

**Comment by yenda** [comment]
Things to test: compatibility with older version of status is still working, especially for commands

**Review by rasom — APPROVED** [review]

**Comment by lukaszfryc** [comment]
> Please do all your testing with special test public/1-1 channels first

@janherich Should there be some specific messages inside those chats? Like empty messages that caused issues in https://github.com/status-im/status-react/issues/6087. Or it's only about testing the PR on some non-visited chats?

**Comment by janherich** [comment]
@lukaszfryc yes, ideally, such messages should be tested as well. However the PR is simply erasing tech debt, so only regressions are relevant.

**Comment by lukaszfryc** [comment]
@janherich I tested 1-1 and public chats (chatting and transacting with 0.9.23, .24 and .29) and I have not found any issues. I also did upgrade tests. Please merge 👍
